src/components/Header/utils/encoreCatalogLogOutTimer.ts

```diff
@@ -0,0 +1,129 @@
+   * have to log in when they are using non-account-required operations
+   * such as searching items. Unfortunately, the patron will also be logged out
+   * from Catalog, so we have to keep tracking the patrons' activites on
+   * Catalog as well. Test Classic Catalog is used for QA.
+   * The default expiration time is 30 mins.
+   */
+  this.setEncoreLoggedInTimer = (currentLocationHost) => {
+    const domainsForExtendingLogIn = [
+      // the domain of Encore's pages
+      "browse.nypl.org",
+      // the domain of Research Catalog's pages
+      // Should this now be "www.nypl.org" since the new catalog is
+      // https://www.nypl.org/research/research-catalog/ ??
+      "catalog.nypl.org",
+      // the domain of Test Research Catalog's pages
+      "nypl-sierra-test.nypl.org",
+    ];
+    const isOnValidDomain = domainsForExtendingLogIn.some(
+      (domain) => domain === currentLocationHost
+    );
+    const isLoggedIn = Cookies.get("PAT_LOGGED_IN");
+
+    if (!isLoggedIn) {
+      // The cookie "PAT_LOGGED_IN" does not exist, delete the
+      // "nyplIdentityPatron" cookie to log the patron out.
+      if (Cookies.get("nyplIdentityPatron")) {
+        Cookies.remove("nyplIdentityPatron");
+      }
+
+      // Delete cookie "VALID_DOMAIN_LAST_VISITED" which holds the last time
+      // the patron visited the valid domain.
+      if (Cookies.get("VALID_DOMAIN_LAST_VISITED")) {
+        Cookies.remove("VALID_DOMAIN_LAST_VISITED");
+      }
+    } else {
+      // Update the "VALID_DOMAIN_LAST_VISITED" cookie in two scenarios:
+      //  1. The patron is on a Sierra (Encore) hosted page, actively
+      //     refresh their session.
+      //  2. The "VALID_DOMAIN_LAST_VISITED" cookie doesn't exist, which will
+      //     only happen if the patron logged in through a redirect, without
+      //     running JS on a "valid domain".
+      if (isOnValidDomain || !Cookies.get("VALID_DOMAIN_LAST_VISITED")) {
+        Cookies.set("VALID_DOMAIN_LAST_VISITED", currentTime);
+        this.logOutFromEncoreAndCatalogIn(patLoggedInCookieExpiredTime);
+      } else {
+        // The time in ms since the patron has been on the valid domain.
+        const lastVisitedValidDomainTime = Cookies.get(
+          "VALID_DOMAIN_LAST_VISITED"
+        );
+        const timeUntilLogOut = lastVisitedValidDomainTime
+          ? patLoggedInCookieExpiredTime -
+            (currentTime - lastVisitedValidDomainTime)
+          : undefined;
+
+        this.logOutFromEncoreAndCatalogIn(timeUntilLogOut);
+      }
+    }
+  };
+
+  /**
+   * The timer to delete logged in related cookies and call the method to
+   * completely log out from Encore and Catalog.
+   */
+  this.logOutFromEncoreAndCatalogIn = (time: number) => {
+    let timeUntilLogOut = time > 0 ? time : 0;
+
+    // When in test mode, set the timeout no longer than 2 seconds.
+    if (isTestMode && timeUntilLogOut > 8000) {
+      timeUntilLogOut = 1000;
```

Comment:
The comment above refers to setting a value to `2 seconds`, but here is it set to 1 second.  Why the descrepancy?